packages/smart-contracts/contracts/Cherryswap.sol

```diff
@@ -1,249 +1,246 @@
-pragma solidity ^0.5.0;
+pragma solidity ^0.5.12;
 
-import "@openzeppelin/upgrades/contracts/Initializable.sol";
-import "@openzeppelin/contracts-ethereum-package/contracts/token/ERC20/ERC20.sol";
+// Library & interfaces
 import "./interface/ICERC20.sol";
 import "./interface/ISwapMath.sol";
+// Contracts
+import "@openzeppelin/upgrades/contracts/Initializable.sol";
+import "@openzeppelin/contracts-ethereum-package/contracts/token/ERC20/ERC20.sol";
+import "./Cherrypool.sol";
+import "./CherryMath.sol";
+
+/**
+ * @title Cherryswap Contract
+ * @dev This contract handle all swaping operations
+ */
+contract Cherryswap is Initializable, Cherrypool {
+    enum Bet {Short, Long}
+
+    uint256 oneMonthDuration = 60 * 60 * 24 * 30;
+    uint256 maxInterestRatePaidPerBlock = uint256((25 * 1e16) / (4 * 60 * 24 * 365)); //25% APR is the max the pool will pay
+
+    uint256 ALPHA = 150; //scaled by 100 so 150 = 1.5
+    uint256 BETA = 0;
+
+    uint256 RAGE_QUITE_PENALTY = 20; //scaled by 100 so 20 = 0.2
+
+    struct Swap {
+        address owner;
+        uint256 swapId;
+        uint256 startingTime;
+        uint256 endingTime;
+        uint256 fixedRateOffer;
+        uint256 depositedValue;
+        uint256 startingcTokenExchangeRate;
+        Bet bet;
+    }
+
+    Swap[] public swaps;
+
+    Cherrymath cherryMath;
+
+    ERC20 token;
+    ICERC20 cToken;
+
```

Comment:
I really think we don't need `ERC20 token` and `ICERC20 cToken` in `Cherryswap.sol``as they exist in the `Cherrypool.sol` and it inherit it.

Reply:
@chrismaree I removed those as we inherit `Cherrypool.sol`